internal/elastic/data_elasticsearch.go

```diff
@@ -99,6 +109,23 @@ func fetchElasticSearchData(ctx context.Context, params *plugin.RetrieveDataPara
 			Detail:   err.Error(),
 		}}
 	}
+	var diags hcl.Diagnostics
+	if (params.Args.GetAttr("only_hits").IsNull() || params.Args.GetAttr("only_hits").True()) &&
+		!params.Args.GetAttr("aggs").IsNull() {
+		if params.Args.GetAttr("query").IsNull() && params.Args.GetAttr("query_string").IsNull() {
+			return nil, hcl.Diagnostics{{
+				Severity: hcl.DiagError,
+				Summary:  "Invalid arguments",
+				Detail:   "Aggregations are not supported without a query or query_string",
```

Comment:
That's not correct -- it should be possible to use `aggr` field without a query / query_string. 

The error should be raised if
- `only_hits` is true
- and `aggs` are set
- and there are no query/query_string

With this combo, `aggs` will not be in the result, while being a primary goal of the data block.

The warning should be raised if
- `only_hits` is true
- and `aggs` are set
- and there IS query/query_string

in this case, while `aggs` are not present in the result, there is valuable data since `hits` are returned

Reply:
Doesn't this error match the case from specs?
> throw an error when:
> only_hits is true, aggs attribute is set, while no query_string or query are set -- the response will contain only aggregations but they will not be returned to the data block since only_hits is true, so there will be no data to use.

Since the if condition is:
```go
if (params.Args.GetAttr("only_hits").IsNull() || params.Args.GetAttr("only_hits").True()) &&
		!params.Args.GetAttr("aggs").IsNull() {
		if params.Args.GetAttr("query").IsNull() && params.Args.GetAttr("query_string").IsNull() {
		
```

Reply:
As for warning, it's added to diagnostics like this:
```go
	diags = diags.Append(&hcl.Diagnostic{
			Severity: hcl.DiagWarning,
			Summary:  "Aggregations are not supported",
			Detail:   "Aggregations are not supported when only_hits is true",
		})
```
Am I missing something here?

Reply:
oh, you're right. apologies, the error message caught my eye but I didn't dig into the condition itself!